enumfields/fields.py

```diff
@@ -34,7 +36,11 @@ def to_python(self, value):
         raise ValidationError('%s is not a valid value for enum %s' % (value, self.enum), code="invalid_enum_value")
 
     def get_prep_value(self, value):
-        return None if value is None else self.enum(value).value
+        if value is None:
+            return None
+        if not isinstance(value, self.enum):
+            return self.enum(value).value
+        return value.value  # Already the correct type
```

Comment:
It's approximately 15% faster on average in [my benchmark](https://gist.github.com/akx/7eec0e49651364a58278919331592579).

Somewhat slower when you pass in a value that needs to be casted (`1` below), but a lot faster when you pass in a value that's already the correct type (`E1.A` below).

`get_prep_value_patch(E1.A)`: 1.267213796993019
`get_prep_value_typecheck(E1.A)`: 0.719976721011335
`get_prep_value_patch(1)` 1.785125483002048
`get_prep_value_typecheck(1)`: 1.8604830430122092

Aggregate results:

`get_prep_value_typecheck`: 2.7982757450372446
`get_prep_value_patch`: 3.270842725993134